beacon-chain/sync/service_test.go

```diff
@@ -3,229 +3,198 @@ package sync
 		exitRoutine <- true
 	}()
 
-	b := types.NewBlock(0)
-	h, err := b.Hash()
-	if err != nil {
-		t.Fatal(err)
+	blockResponse := pb.BeaconBlockResponse{
+		MainChainRef: []byte("hi"),
 	}
 
-	// if the hash and the block are processed in order
-	ss.ReceiveBlockHash(h)
-	ss.ReceiveBlock(b)
+	msg := p2p.Message{
+		Peer: p2p.Peer{},
+		Data: blockResponse,
+	}
+
+	ss.blockBuf <- msg
 	ss.cancel()
 	<-exitRoutine
 
-	// sync service broadcasts the block and forwards the block to to the local chain
-	testutil.AssertLogsContain(t, hook, fmt.Sprintf("broadcasting block: %x", h.Sum(nil)))
-	testutil.AssertLogsContain(t, hook, fmt.Sprintf("forwarding block: %x", h.Sum(nil)))
+	// Sync service broadcasts the block and forwards the block to to the local chain.
+	testutil.AssertLogsContain(t, hook, "Broadcasting block hash to peers")
+	testutil.AssertLogsContain(t, hook, "Processed block")
+	hook.Reset()
 }
 
 func TestProcessMultipleBlocks(t *testing.T) {
 	hook := logTest.NewGlobal()
 
 	cfg := Config{HashBufferSize: 0, BlockBufferSize: 0}
-	ss := NewSyncService(context.Background(), cfg)
-
-	ns := MockNetworkService{}
-	cs := MockChainService{}
-
-	ss.SetNetworkService(&ns)
-	ss.SetChainService(&cs)
+	ss := NewSyncService(context.Background(), cfg, &mockP2P{}, &mockChainService{})
 
 	exitRoutine := make(chan bool)
 
 	go func() {
-		run(ss.ctx.Done(), ss.hashBuf, ss.blockBuf, &ns, &cs)
+		ss.run(ss.ctx.Done())
 		exitRoutine <- true
 	}()
 
-	b1 := types.NewBlock(0)
-	h1, err := b1.Hash()
-	if err != nil {
-		t.Fatal(err)
+	blockResponse1 := pb.BeaconBlockResponse{
+		MainChainRef: []byte("foo"),
 	}
 
-	b2 := types.NewBlock(1)
-	h2, err := b2.Hash()
-	if err != nil {
-		t.Fatal(err)
+	msg1 := p2p.Message{
+		Peer: p2p.Peer{},
+		Data: blockResponse1,
 	}
 
-	if bytes.Equal(h1.Sum(nil), h2.Sum(nil)) {
-		t.Fatalf("two blocks should not have the same hash:\n%x\n%x", h1.Sum(nil), h2.Sum(nil))
+	blockResponse2 := pb.BeaconBlockResponse{
+		MainChainRef: []byte("bar"),
 	}
 
-	// if two different blocks are submitted
-	ss.ReceiveBlockHash(h1)
-	ss.ReceiveBlock(b1)
-	ss.ReceiveBlockHash(h2)
-	ss.ReceiveBlock(b2)
+	msg2 := p2p.Message{
+		Peer: p2p.Peer{},
+		Data: blockResponse2,
+	}
+
+	ss.blockBuf <- msg1
+	ss.blockBuf <- msg2
 	ss.cancel()
 	<-exitRoutine
 
-	// both blocks are processed
-	testutil.AssertLogsContain(t, hook, fmt.Sprintf("broadcasting block: %x", h1.Sum(nil)))
-	testutil.AssertLogsContain(t, hook, fmt.Sprintf("forwarding block: %x", h1.Sum(nil)))
-	testutil.AssertLogsContain(t, hook, fmt.Sprintf("broadcasting block: %x", h2.Sum(nil)))
-	testutil.AssertLogsContain(t, hook, fmt.Sprintf("forwarding block: %x", h2.Sum(nil)))
+	// Sync service broadcasts the two separate blocks
+	// and forwards them to to the local chain.
+	testutil.AssertLogsContain(t, hook, "Broadcasting block hash to peers")
+	testutil.AssertLogsContain(t, hook, "Processed block")
+	testutil.AssertLogsContain(t, hook, "Broadcasting block hash to peers")
```

Comment:
This doesn't test that two different blocks were processed. A potential fix is to include the hash in `BeaconBlockResponse, log the hash, and update this assertion.